distributed/comm/tcp.py

```diff
@@ -367,8 +367,24 @@ def _check_encryption(self, address, connection_args):
 
 class BaseTCPConnector(Connector, RequireEncryptionMixin):
     _executor = ThreadPoolExecutor(2, thread_name_prefix="TCP-Executor")
-    _resolver = netutil.ExecutorResolver(close_executor=False, executor=_executor)
-    client = TCPClient(resolver=_resolver)
+
+    @property
```

Comment:
Wouldn't this set an instance attribute, so you get a new Resolver for every instance? Previously, by having it be on the class, there is only ever one for all instances of this class and descendants. You would need a [class property](https://stackoverflow.com/a/5191224) or descriptor for the same effect, or have class attribute `_cached_resolver = [None]`, above, as in other cases.

Reply:
Or I guess just `BaseTCPConnector._cached_resolver = ` in the Except block